internal/pkg/service/stream/migrate/migrate.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"sync"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/migrate/source"
+)
+
+type Flags struct {
+	storageAPIToken string
+	host            string
+}
+
+func NewFlags() Flags {
+	storageAPIToken := flag.String("storage-api-token", "", "storage api token")
+	host := flag.String("host", "", "host")
+
+	flag.Parse()
+
+	return Flags{*storageAPIToken, *host}
+}
+
+func main() {
+	ctx := context.Background()
+	logger := log.NewServiceLogger(os.Stdout, true)
+	wg := sync.WaitGroup{}
+
+	flags := NewFlags()
+	if flags.host == "" && flags.storageAPIToken == "" {
+		logger.Error(ctx, "usage: migrate storage-api-token host")
+		os.Exit(1)
+	}
+
+	// fetch receivers and exports from old API
+	bufferReceivers, err := source.FetchBufferReceivers(ctx, flags.host, flags.storageAPIToken, logger)
+	if err != nil {
+		logger.Error(ctx, err.Error())
+		os.Exit(1)
+	}
+
+	logger.Info(ctx, `Starting migration...`)
+	for _, receiver := range bufferReceivers.Receivers {
+		wg.Add(1)
+		go func() {
+			err = receiver.CreateSource(ctx, flags.storageAPIToken, flags.host, &wg, logger)
+			if err != nil {
+				logger.Error(ctx, err.Error())
+			}
+		}()
```

Comment:
Why the `defer wg.Done()` is not here but in function.  Since you use it as prerequisite step as first statement in `CreateSource` I would move it here as first step.

Reply:
After discussion with Michal the goroutine was removed,because there is not so much data.

Reply:
[909458d](https://github.com/keboola/keboola-as-code/pull/1743/commits/909458d63ae43159c11dd8a067cab9016a6f24bb)